### DISCUSSION TITLE
Pagination on tables with Mongodb

### DISCUSSION THREAD

**Discussion by introvert** [Budibase/budibase#7332] [Help] [answered]
I'm new to Budibase and noticed that by default Budibase doesn't support table pagination with Mongodb as it requires to define queries.

Is there an example how to support pagination on tables when using it?

**Comment by melohagan** [comment]
Hi @introvert 

Unfortunately pagination is not supported for MongoDB or other custom queries for that matter. There's a couple of related issues here:
- https://github.com/Budibase/budibase/issues/6226
- https://github.com/Budibase/budibase/issues/3974

**Comment by introvert** [comment]
This is a big downside of Budibase. Are there any plans to address this or is there any other solution to it?

**Comment by melohagan** [comment, marked as answer]
Hey @introvert 

Pagination isn't possible in the MongoDB connector yet as we need to add support for **limit**.

Another consideration is thinking about how to handle pagination. 
Looking at this article <a href="https://www.codementor.io/@arpitbhayani/fast-and-efficient-pagination-in-mongodb-9095flbqr">here</a> using ObjectId and limit is probably the way to go over using skip.

Another good post here: https://medium.com/swlh/mongodb-pagination-fast-consistent-ece2a97070f3

I've raised a feature request here: https://github.com/Budibase/budibase/issues/7355
